eo-maven-plugin/src/main/java/org/eolang/maven/ProbeMojo.java

```diff
@@ -0,0 +1,250 @@
+                ftojo.set(AssembleMojo.ATTR_DISCOVERED_AT, src);
+                probed.add(name);
+            }
+            tojo.set(AssembleMojo.ATTR_HASH, new ChNarrow(hash).value());
+            tojo.set(AssembleMojo.ATTR_PROBED, Integer.toString(count));
+        }
+        if (tojos.isEmpty()) {
+            if (this.scopedTojos().select(row -> true).isEmpty()) {
+                Logger.warn(this, "Nothing to probe, since there are no programs");
+            } else {
+                Logger.info(this, "Nothing to probe, all programs checked already");
+            }
+        } else if (probed.isEmpty()) {
+            Logger.info(
+                this, "No probes found in %d programs",
+                tojos.size()
+            );
+        } else {
+            Logger.info(
+                this, "Found %d probes in %d programs: %s",
+                probed.size(), tojos.size(), probed
+            );
+        }
+    }
+
+    /**
+     * Find all probes found in the provided XML file.
+     *
+     * @param file The .xmir file
+     * @return List of foreign objects found
+     * @throws FileNotFoundException If not found
+     * @todo #1395:30min Rewrite lines 187-204 as fully `cactoos` style to
+     *  make to code more convenient. So there is will no "imperative" `forEach()`
+     *  in line 194.
+     */
+    private Collection<String> probes(final Path file)
+        throws FileNotFoundException {
+        final XML xml = new XMLDocument(file);
+        final Collection<String> ret = new HashSet<>(1);
+        new TreeSet<>(
+            new ListOf<>(
+                new Filtered<>(
+                    obj -> !obj.isEmpty(),
+                    xml.xpath("//metas/meta[head/text() = 'probe']/tail/text()")
+                )
+            )
+        ).forEach(
+            obj -> {
+                if (!ProbeMojo.hasReservedChars(obj)) {
+                    if (obj.length() > 1 && "Q.".equals(obj.substring(0, 2))) {
+                        ret.add(obj.substring(2));
+                    } else {
+                        ret.add(obj);
+                    }
+                }
+            }
+        );
+        if (ret.isEmpty()) {
+            Logger.debug(
+                this, "Didn't find any probed objects in %s",
+                new Rel(file)
+            );
+        } else {
+            Logger.debug(
+                this, "Found %d probed objects in %s: %s",
+                ret.size(), new Rel(file), ret
+            );
+        }
+        return ret;
+    }
+
+    /**
+     * Checks if String has reserved symbols.
+     *
+     * @param str String
+     * @return True if found
+     * @todo #1395:30min Need to add the logic of "hasReservedChars" method to
+     *  add-probes.xsl". After that, the method in this class need to be removed.
+     * @checkstyle BooleanExpressionComplexityCheck (15 lines)
+     */
+    private static boolean hasReservedChars(final String str) {
+        return str.contains("<")
```

Comment:
@MikhailLipanin I believe we can replace that code with:
```java
return Stream.of("<",">","$","*", "?", ":", "\"", "|", "^", "@").anyMatch(str::contains);
```
In that case we don't need to suppress warning `@checkstyle BooleanExpressionComplexityCheck (15 lines)`

Reply:
done